src/interface.ts

```diff
@@ -72,7 +104,23 @@ interface DesktopAgent {
    * Returns a promise that resolves to an Array. The resolved dataset & metadata is Desktop Agent-specific.
    * If the resolution errors, it returns an `Error` with a string from the `ResolveError` enumeration.
    */
-  resolve(intent: IntentName, context: Context): Promise<Array<AppMetadata>>;
+  resolveIntent(intent: Intent): Promise<ResolveIntentResults>;
+
+  /**
+   * Resolves a context to a list of intents supported by the subsystem.
+   *
+   * ResolveContext is effectively granting programmatic access to the Desktop Agent's resolver. 
+   * Returns a promise of type `ResolveContextResults`.  The resolved list of intents is Desktop Agent-specific.
+   * If the resolution errors, it returns an `Error` with a string from the `ResolveError` enumeration.
+   */
+  resolveContext(context: Context): Promise<ResolveContextResults>;
+
+  /**
+   * sends an intent, containing a context and optional target
+   * 
+   * If firing errors, it returns an `Error` with a string from the `ResolveError` or `OpenError` enumerations.
+   */
+  raiseIntent(intent: Intent): Promise<RaiseIntentResults>;
```

Comment:
Minor: I know this is just semantics/naming, and I am slightly splitting hairs, but I really dislike "raise" for an intent. I would like us to stick with the previous agreed naming of "sending" an intent. You raise events, not intents. An intent is something that you "express", if you will.

Reply:
I think naming/semantics are important, hence why I changed *raise*.  I don't like send because it implies a receiver which isn't really the case here.  Happy to change, but send is misleading in my opinion.

Reply:
I see your point (although I don't agree that sending something presumes there is a receiver on the other side). I am not sure raise is better than send, though, it is misleading in its own way. Maybe `broadcastIntent` like Android uses is best?

Happy to go with the majority opinion on this one.